pkg/proxy/proxy.go

```diff
@@ -159,6 +160,119 @@ func (p *SprayProxy) HandleProxy(c *gin.Context) {
 	c.String(http.StatusOK, "proxied")
 }
 
+func (p *SprayProxy) HandleProxyEndpoint(c *gin.Context) {
+	// currently not distinguishing between requests we can parse and those we cannot parse
+	metrics.IncInboundCount()
+	errors := []error{}
+	zapCommonFields := []zapcore.Field{
+		zap.String("method", c.Request.Method),
+		zap.String("path", c.Request.URL.Path),
+		zap.String("query", c.Request.URL.RawQuery),
+		zap.Bool("insecure-tls", p.insecureTLS),
+		zap.String("request-id", c.GetString("requestId")),
+	}
+	// Read in body from incoming request
+	buf := &bytes.Buffer{}
+	_, err := buf.ReadFrom(c.Request.Body)
+	defer c.Request.Body.Close()
```

Comment:
Need to rebase and include #43 logic.

You can also treat this as an opportunity to create a separate function that handles the core proxying logic. This is known as DRYing out your code (Don't Repeat Yourself).

Reply:
Done